packages/sdk/client-services/src/packlets/logging/logging-service.ts

```diff
@@ -0,0 +1,55 @@
+//
+// Copyright 2022 DXOS.org
+//
+
+import { Event } from '@dxos/async';
+import { Stream } from '@dxos/codec-protobuf';
+import { log, LogEntry as NaturalLogEntry, LogProcessor, shouldLog } from '@dxos/log';
+import { LoggingService, LogEntry, QueryLogsRequest } from '@dxos/protocols/proto/dxos/client/services';
+
+/**
+ * Logging service used to spy on logs of the host.
+ */
+export class LoggingServiceImpl implements LoggingService {
+  private readonly _active = new Set<Event<LogEntry>>();
```

Comment:
why not
```suggestion
  private readonly _log = new Event<LogEntry>();
```

and then the stream subscribes to the even